wwwbase/ajax/scramble.php

```diff
@@ -0,0 +1,201 @@
+  $lexem = "";
+  $lexem = randomWordGenerator();  
+  $letters = str_split_unicode($lexem);
+  gen($letters, 0, $results);
+  //print_r($results);
+
+  $wordList = array();
+  $wordList = findWords($results);
+  //print_r($wordList);
+  ajaxEcho($lexem, $wordList);
+
+  //print_r($results);
+
+  //for spliting into substrings a word of 5 letters
+  /*
+  for ($i = 0; $i < 3; $i++) {
+      $subWord = mb_substr($lexem, $i, 3);
+      $lexemArray = str_split_unicode($subWord);
+      $temp = letterPermute($lexemArray);
+      $wordList = array_merge($wordList, $temp);
+    }
+//echo '<pre>'; print_r($wordList); echo '</pre>';
+  for ($i = 0; $i < 2; $i++) {
+    $subWord = mb_substr($lexem, $i, 4);
+    $lexemArray = str_split_unicode($subWord);
+    $temp = letterPermute($lexemArray);
+    $wordList = array_merge($wordList, $temp);
+  }
+//first char + last 2 chars
+  for($i = 0; $i < 2; $i++) {
+    $subWord = mb_substr($lexem, $i, 1); 
+    $subWord .= mb_substr($lexem, 3, 2);
+    $lexemArray = str_split_unicode($subWord);
+    $temp = letterPermute($lexemArray);
+    $wordList = array_merge($wordList, $temp);
+  }
+  //first char + last 3 chars
+  $subWord = mb_substr($lexem, 0, 1);
+  $subWord .= mb_substr($lexem, 2, 3);
+  $lexemArray = str_split_unicode($subWord);
+  $temp = letterPermute($lexemArray);
+  $wordList = array_merge($wordList, $temp);
+  //first 3 chars + last char
+  $subWord = mb_substr($lexem, 0, 3);
+  $subWord .= mb_substr($lexem, 4, 1);
+  $lexemArray = str_split_unicode($subWord);
+  $temp = letterPermute($lexemArray);
+  $wordList = array_merge($wordList, $temp);
+  //permute the whole word.
+  $lexemSplit = $lexem;
+  $lexemArray = str_split_unicode($lexemSplit);
+  $wordList = array_merge($wordList, letterPermute($lexemArray));
+
+  $wordArray = findWords($wordList);
+  ajaxEcho($lexem,$wordArray);
+  */
+
+}
+
+// Permută pozițiile $k ... |$v|-1
+function gen(&$v, $k, &$results) {
+  // Primele $k elemente constituie o soluție (peste o limită minimă)
+  if ($k >= MIN_LENGTH) {
+    $results[] = implode(array_slice($v, 0, $k));
+  }
+
+  // Dacă n-am epuizat vectorul, continui
+  if ($k < count($v)) {
+    // Pe poziția $k pun, pe rând, fiecare din elementele rămase
+    for ($i = $k; $i < count($v); $i++) {
+      $tmp = $v[$k];
+      $v[$k] = $v[$i];
+      $v[$i] = $tmp;
+
+      // Reapelez de la poziția $k + 1
+      gen($v, $k + 1, $results);
+
+      // Apoi pun elementul inițial la loc
+      $tmp = $v[$k];
+      $v[$k] = $v[$i];
+      $v[$i] = $tmp;
+    }
+  }
+  // Astfel, la final, $v este nemodificat
+}
+
+
+function findWords($wordList) {
+  //$wordList1 = array();
+  $wordsFound = array(); //the final words that were found after doing all the possible combinations.
+  $j = 0;
+  for($i = 0 ;$i < count($wordList); $i++) {
+    //$wordList1[$i] = implode($wordList[$i]);
+    $search = Model::factory('Lexem')
+            ->where("formUtf8General", $wordList[$i])
+            ->find_one();
```

Comment:
Scuze pentru întârziere. DEX, DOOM și DOR sunt dicționare. Lor le corespund înregistrări din tabela `Source`, de exemplu id = 1 este DEX 1998, id = 27 este DEX 2009... Nu vrem să căutăm în toate sursele, căci multe sunt arhaice, regionale etc. Dar mi-a venit o idee mai bună ca să nu facem 5-6 joinuri ca să putem filtra după Source.id, căci ar fi lent. Vezi mai jos.

Apoi: `Lexem` este tabela de cuvinte (forma de bază, partea de vorbire etc.). Dar jocul trebuie să permită și toate formele flexionare (deci nu doar _apă_, ci și _ape_, _apei_, _apelor_ etc.). Acestea sunt stocate în tabela `InflectedForm`.

Mi-a venit ideea să acceptăm formele care sunt acceptate la jocul de scrabble. Acestea sunt un subset bun, fără regionalisme obscure. Pentru asta, interogarea ar fi:

```
select * from InflectedForm i join LexemModel lm on i.lexemModelId = lm.id where i.formUtf8General = 'apă' and lm.isLoc limit 1;
```
Iar interogarea cu paris/idiorm (sper să se compileze, n-am testat-o):

$search = Model::factory('InflectedForm')
  ->select('i.*')
  ->table_alias('i')
  ->join('LexemModel', ('i.lexemModelId', '=', 'lm.id'), 'lm')
  ->where('i.formUtf8General', $wordList[$i])
  ->where('lm.isLoc', 1)
  ->find_one();

Pentru clarificare, căutările după `formUtf8General` ignoră diacriticele (deci A = Ă = Â), iar căutările după `formNoAccent` țin cont de diacritice. Încă nu știu ce decizie ai luat, dar îți recomand să ignori toate diacriticele. Așa este și la scrabble, rebus etc.